### ISSUE TITLE
[Documentation] 

### ISSUE THREAD

**Issue by teslyuk** [microsoft/onnxruntime#14859]
### Wide character type PyTorch Inference crash

I am using ONNX Runtime C++ pre-built version 1.14.0 and following along with tutorial on PyTorch Inference. While I'm trying to build inference example I'm getting error using wide character type, line:

`Ort::Session session{env, L"model.onnx", Ort::SessionOptions{nullptr}};`

Crashing with error: 
```
error: no matching function for call to ‘Ort::Session::Session(<brace-enclosed initializer list>)’
   14 | v, L"/home/novarc/onnxruntime-example/resnest_50_new_aug_ur_1.onnx", Ort::SessionOptions{nullptr}};
      |                                                                                                  ^

In file included from /opt/onnxruntime/include/onnxruntime_cxx_api.h:1876,
                 from /home/novarc/onnxruntime-example/main.cpp:5:
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:950:8: note: candidate: ‘Ort::Session::Session(const Ort::Env&, const void*, size_t, const Ort::SessionOptions&, OrtPrepackedWeightsContainer*)’
  950 | inline Session::Session(const Env& env, const void* model_data, size_t model_data_length,
      |        ^~~~~~~
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:950:8: note:   candidate expects 5 arguments, 3 provided
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:946:8: note: candidate: ‘Ort::Session::Session(const Ort::Env&, const void*, size_t, const Ort::SessionOptions&)’
  946 | inline Session::Session(const Env& env, const void* model_data, size_t model_data_length, const SessionOptions& options) {
      |        ^~~~~~~
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:946:8: note:   candidate expects 4 arguments, 3 provided
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:941:8: note: candidate: ‘Ort::Session::Session(const Ort::Env&, const char*, const Ort::SessionOptions&, OrtPrepackedWeightsContainer*)’
  941 | inline Session::Session(const Env& env, const ORTCHAR_T* model_path, const SessionOptions& options,
      |        ^~~~~~~
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:941:8: note:   candidate expects 4 arguments, 3 provided
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:937:8: note: candidate: ‘Ort::Session::Session(const Ort::Env&, const char*, const Ort::SessionOptions&)’
  937 | inline Session::Session(const Env& env, const ORTCHAR_T* model_path, const SessionOptions& options) {
      |        ^~~~~~~
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:937:58: note:   no known conversion for argument 2 from ‘const wchar_t [62]’ to ‘const char*’
  937 | inline Session::Session(const Env& env, const ORTCHAR_T* model_path, const SessionOptions& options) {
      |                                         ~~~~~~~~~~~~~~~~~^~~~~~~~~~
In file included from /home/novarc/onnxruntime-example/main.cpp:5:
/opt/onnxruntime/include/onnxruntime_cxx_api.h:783:12: note: candidate: ‘Ort::Session::Session(std::nullptr_t)’
  783 |   explicit Session(std::nullptr_t) {}                                                   ///< Create an empty Session object, must be assigned a valid one to be used
      |            ^~~~~~~
/opt/onnxruntime/include/onnxruntime_cxx_api.h:783:12: note:   candidate expects 1 argument, 3 provided
/opt/onnxruntime/include/onnxruntime_cxx_api.h:782:8: note: candidate: ‘Ort::Session::Session(Ort::Session&&)’
  782 | struct Session : detail::SessionImpl<OrtSession> {
      |        ^~~~~~~
/opt/onnxruntime/include/onnxruntime_cxx_api.h:782:8: note:   candidate expects 1 argument, 3 provided
make[2]: *** [CMakeFiles/onnxruntime-example.dir/build.make:76: CMakeFiles/onnxruntime-example.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/onnxruntime-example.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

If I'm changing this line to: `Ort::Session session{env, "model.onnx", Ort::SessionOptions{nullptr}};` everything works fine.

### Page / URL

https://onnxruntime.ai/docs/tutorials/accelerate-pytorch/pytorch.html

**Comment by fs-eire** [comment]
This is the expected behavior. ORTCHAR_T is defined as wide char in windows but char in UNIX

**Comment by natke** [comment]
Fixed by #14872
